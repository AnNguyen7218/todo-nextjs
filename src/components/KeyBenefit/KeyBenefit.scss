@import 'config.gutters';
@import 'config.colors';
@import 'config.media';

$benefit-icon-width-sm: 2.4rem;
$benefit-icon-width-xl: 4.8rem;

.container {
  display: grid;
  grid-template-columns: $benefit-icon-width-sm 1fr;
  column-gap: gutter(16);
  grid-template-areas:
    'icon title'
    'empty description';
  margin-bottom: gutter(24);
  row-gap: gutter(4);

  @include break(xl) {
    grid-template-areas:
      'icon title'
      'icon description';
    grid-template-columns: $benefit-icon-width-xl 1fr;
    column-gap: gutter(36);
    margin-bottom: gutter(48);
  }
}

.icon {
  width: $benefit-icon-width-sm;
  height: $benefit-icon-width-sm;
  color: $purple;
  grid-area: icon;
  align-self: center;

  @include break(xl) {
    width: $benefit-icon-width-xl;
    height: $benefit-icon-width-xl;
    align-self: flex-start;
  }
}

.title {
  grid-area: title;
  align-self: center;

  @include break(xl) {
    align-self: flex-start;
  }
}

.description {
  grid-area: description;
  align-self: center;

  @include break(xl) {
    min-width: 28.4rem;
  }

  @include break(xl) {
    align-self: flex-start;
  }
}
