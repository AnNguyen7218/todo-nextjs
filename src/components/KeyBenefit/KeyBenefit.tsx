import React from 'react';
import css from './KeyBenefit.scss';

type Props = {
  benefit: KeyBenefitType;
};

const KeyBenefit: React.FC<Props> = ({ benefit }) => {
  const { icon: Icon, title, description } = benefit;
  return (
    <div className={css['container']}>
      <Icon className={css['icon']} />
      <h3 className={css['title']}>{title}</h3>
      <p className={css['description']}>{description}</p>
    </div>
  );
};

export default KeyBenefit;
