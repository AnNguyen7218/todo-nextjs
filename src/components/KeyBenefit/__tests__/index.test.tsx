import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Icon from 'assets/icons/ic-cashflow.svg';
import KeyBenefit from '../KeyBenefit';

afterEach(cleanup);

const ITEM = {
  icon: Icon,
  title: 'Item',
  description: 'Item description',
};

describe('KeyBenefit Component', () => {
  test('KeyBenefit should match snapshot', () => {
    const { container } = render(<KeyBenefit benefit={ITEM} />);
    expect(container).toMatchSnapshot();
  });
});
