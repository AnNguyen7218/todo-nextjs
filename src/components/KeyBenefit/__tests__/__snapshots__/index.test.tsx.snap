// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyBenefit Component KeyBenefit should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <ic-cashflow.svg
      classname="icon"
    />
    <h3
      class="title"
    >
      Item
    </h3>
    <p
      class="description"
    >
      Item description
    </p>
  </div>
</div>
`;
